tests/integration/metrics/column_values/test_non_null.py

```diff
@@ -0,0 +1,131 @@
+from typing import Sequence
+
+import pandas as pd
+
+from great_expectations.compatibility.pyspark import functions as F
+from great_expectations.compatibility.sqlalchemy import BinaryExpression, ColumnClause
+from great_expectations.metrics.column_values.non_null import (
+    ColumnValuesNonNull,
+    ColumnValuesNonNullCount,
+    ColumnValuesNonNullCountResult,
+    ColumnValuesNonNullResult,
+)
+from tests.integration.conftest import parameterize_batch_for_data_sources
+from tests.integration.test_utils.data_source_config import (
+    BigQueryDatasourceTestConfig,
+    DatabricksDatasourceTestConfig,
+    DataSourceTestConfig,
+    MSSQLDatasourceTestConfig,
+    PandasDataFrameDatasourceTestConfig,
+    PandasFilesystemCsvDatasourceTestConfig,
+    PostgreSQLDatasourceTestConfig,
+    SnowflakeDatasourceTestConfig,
+    SparkFilesystemCsvDatasourceTestConfig,
+    SqliteDatasourceTestConfig,
+)
+
+STRING_COLUMN_NAME = "letter"
+DATA_FRAME = pd.DataFrame(
+    {
+        STRING_COLUMN_NAME: ["a", None, "c", "d"],
+    },
+)
+
+PANDAS_DATA_SOURCES: Sequence[DataSourceTestConfig] = [
+    PandasFilesystemCsvDatasourceTestConfig(),
+    PandasDataFrameDatasourceTestConfig(),
+]
+
+SPARK_DATA_SOURCES: Sequence[DataSourceTestConfig] = [
+    SparkFilesystemCsvDatasourceTestConfig(),
+]
+
+SQL_DATA_SOURCES: Sequence[DataSourceTestConfig] = [
+    BigQueryDatasourceTestConfig(),
+    DatabricksDatasourceTestConfig(),
+    MSSQLDatasourceTestConfig(),
+    PostgreSQLDatasourceTestConfig(),
+    SnowflakeDatasourceTestConfig(),
+    SqliteDatasourceTestConfig(),
+]
```

Comment:
i suggest we move these to a common `conftest` module so we can test every metric against a standard set of backends.

We should probably restrict this list to only test against data sources which are officially supported by core. As far as I know, MSSQL is not on that list.